### DISCUSSION TITLE
Facing Error error @azure-tools/typespec-azure-resource-manager/template-type-constraint-no-met

### DISCUSSION THREAD

**Discussion by kapdrish** [microsoft/typespec#3904] [Q&A]
I'm trying to add an extension resource named approvals.
Here are my changes [https://dev.azure.com/msazuredev/VirtualEnclaves/_git/ve-common/commit/2b8213089d00928031792d571aa6fc92cfe57cbc?refName=refs%2Fheads%2Fdrkapoor%2FtypespecChanges](https://dev.azure.com/msazuredev/VirtualEnclaves/_git/ve-common/commit/2b8213089d00928031792d571aa6fc92cfe57cbc?refName=refs%2Fheads%2Fdrkapoor%2FtypespecChanges). I keep facing this error when i compile the changes, `@azure-tools/typespec-azure-resource-manager/template-type-constraint-no-met` 

![image](https://github.com/user-attachments/assets/af2741d6-b78d-4b51-a2de-0decf7e04638)

Needed help figuring out where I'm going wrong. Thanks

**Comment by timotheeguerin** [comment]
For question about azure libraries please ask them in the Teams channel or file bug in this repo https://github.com/Azure/typespec-azure.

This error tells you the resource you are passing is not an arm resource. It should extend on of the TrackedResource, ProxyResource, etc. See [docs](https://azure.github.io/typespec-azure/docs/howtos/ARM/resource-type)
